### PR TITLE
[global] add isort and `make format`

### PR DESCRIPTION
**Pull request by david-caro**
**What:**

* Create a new command `make format` to apply all the formatters.
* Add isort formatter
* Actually apply all the formatters
* Also moved all the relative import to non-relative when possible (just
  standardizing on something).

**Why:**

My editor kept autoformatting the code every time I touched it, applying isort
also, this standardizes it and gives a helpful command to ease the formatting.

**How:**

Added isort test rule to the makefile, and a new `format` rule that runs both
black and isort to reformat the code.

**Risks:**

Little, maybe someone with local code might have some merge conflict.

**Checklist**:

- [x] Added tests, if you've added code that should be tested
- [x] Updated the documentation, if you've changed APIs
- [x] Ensured the test suite passes
- [x] Made sure your code lints
- [x] Completed the Contributor License Agreement ("CLA")

### CODE REVIEW

**Comment by david-caro** [comment]
I'm not 100% sure, but I think that the decrease in coverage comes from the refactoring of the imports.

**Comment by fabriziocucci** [comment]
Apologies for the delay @david-caro, I will look at it today, seems promising! :)

**Comment by fabriziocucci** [comment]
Hey @david-caro, I was documenting myself on the subject and:

1. [optimizing imports is coming to black](https://github.com/psf/black/issues/333) (but who knows when)
2. have you considered using the [isort black profile](https://timothycrosley.github.io/isort/docs/configuration/profiles/)? If yes/no, why?

There rest looks very neat, happy to approve after this! 👍

**Comment by david-caro** [comment]
> 1. [optimizing imports is coming to black](https://github.com/psf/black/issues/333) (but who knows when)

Crossing fingers xd, if I have time I might try to give it a try and help :)

> 2. have you considered using the [isort black profile](https://timothycrosley.github.io/isort/docs/configuration/profiles/)? If yes/no, why?

I did not see it, I did saw the issue and I don't remember any mention of profiles... Xd  I'll be more than happy to change it!

**Comment by ichard26** [comment]
Black contributor here! So a lot changed since the opening of https://github.com/psf/black/issues/333. It appears that we have [deferred to the author of isort for improving the situation of using Black and isort together](https://github.com/psf/black/issues/333#issuecomment-490245892): 
> `black_isort` written by the original author of isort sounds like the ideal situation for us. Thanks so much for your input. Let me know if you need any help.

Then, isort 5 landed and with it came many changes. Including profiles which in essence are names attached to pre-defined sets of configurations. Helpfully, there is a Black profile available. Although, isort 5 cannot run on Python3.5 and below so please take that into account (I stumbled across this issue and project so I barely have a idea with this projects is even about :upside_down_face:). 

For more information:
https://timothycrosley.github.io/isort/docs/major_releases/introducing_isort_5/
https://github.com/psf/black/blob/master/docs/compatible_configs.md#isort

File: tests/cli_unittest.py [hunk 1]
```diff
@@ -559,24 +559,25 @@ def test_dsl_debug(self):
         self.argv.append("--dsl-debug")
         self.run_testslide(
             expected_return_code=1,
-            expected_stdout_startswith=(
+            expected_regex_in_stdout=(
                 "top context\n"
-                "  example: passing_example @ tests/sample_tests.py:34\n"
+                "  example: passing_example @ tests/sample_tests.py:\d+\n"
```
Comment (fornellas) [inline tests/cli_unittest.py:564]: Thanks for fixing this!! It have been bugging all of us for a while!

**Pull request merged**